flexmeasures/data/tests/test_scheduling_repeated_jobs.py

```diff
@@ -0,0 +1,331 @@
+        "key2": "value2",
+        "key3": 3,
+        "key4": {"key1_nested": 1, "key2_nested": 2},
+        "key5": '{"serialized_key1_nested": 1, "serialized_key2_nested": 2}',
+    }
+
+    hash_original = hash_function_arguments(args, kwargs)
+    hash_modified = hash_function_arguments(args_modified, kwargs_modified)
+
+    if equal:
+        assert hash_original == hash_modified
+    else:
+        assert hash_original != hash_modified
+
+
+def test_hashing(db, app, add_charging_station_assets, setup_test_data):
+    soc_at_start = 1
+    target_soc = 5
+    duration_until_target = timedelta(hours=2)
+
+    charging_station = Sensor.query.filter(
+        Sensor.name == "Test charging station"
+    ).one_or_none()
+    tz = pytz.timezone("Europe/Amsterdam")
+    start = tz.localize(datetime(2015, 1, 2))
+    end = tz.localize(datetime(2015, 1, 3))
+    target_datetime = start + duration_until_target
+    resolution = timedelta(minutes=15)
+    soc_targets = [dict(datetime=target_datetime.isoformat(), value=target_soc)]
+
+    kwargs = dict(
+        sensor=charging_station,
+        start=start,
+        end=end,
+        belief_time=start,
+        resolution=resolution,
+        flex_model={"soc-at-start": soc_at_start, "soc-targets": soc_targets},
+    )
+    args = []
+
+    hash = hash_function_arguments(args, kwargs)
+    print(hash)
+
+    # checks that hashes are consistent between different runtime calls
+    assert hash == "+aVb5DY1c64pu7wHl8XHvmbClu6Y9fqww8QDOrlrtCM="
+
+    kwargs2 = copy.deepcopy(kwargs)
+    args2 = copy.deepcopy(args)
+
+    # checks that hashes are consistent within the same runtime calls
+    hash2 = hash_function_arguments(args2, kwargs2)
+    assert hash2 == hash
+
+    # checks that different arguments yield different hashes
+    kwargs2["resolution"] = timedelta(minutes=12)
+    hash3 = hash_function_arguments(args2, kwargs2)
+    assert hash != hash3
+
+
+def test_scheduling_multiple_triggers(
+    caplog, db, app, add_charging_station_assets, setup_test_data
+):
+    caplog.set_level(
+        logging.INFO
+    )  # setting the logging level of the log capture fixture
+
+    soc_at_start = 1
+    target_soc = 5
+    duration_until_target = timedelta(hours=2)
+
+    charging_station = Sensor.query.filter(
+        Sensor.name == "Test charging station"
+    ).one_or_none()
+    tz = pytz.timezone("Europe/Amsterdam")
+    start = tz.localize(datetime(2015, 1, 2))
+    end = tz.localize(datetime(2015, 1, 3))
+    target_datetime = start + duration_until_target
+    resolution = timedelta(minutes=15)
+    soc_targets = [dict(datetime=target_datetime.isoformat(), value=target_soc)]
+
+    assert (
+        DataSource.query.filter_by(name="FlexMeasures", type="scheduling script")
+        .where()
+        .one_or_none()
+        is None
+    )  # Make sure the scheduler data source isn't there
+
+    # schedule 1 job
+    job1 = create_scheduling_job(
+        sensor=charging_station,
```

Comment:
This code to create jobs 1 through 3 can be shorter if we make a dict `scheduling_args` first, then use it three times. One time with a modification.

Reply:
Oh, good point. I'll change this to make the code shorter (+easy to follow).